dali/python/nvidia/dali/plugin/mxnet.py

```diff
@@ -51,7 +59,74 @@ def feed_ndarray(dali_tensor, arr):
     # Copy data from DALI tensor to ptr
     dali_tensor.copy_to_external(ptr)
 
-class DALIGenericIterator(mx.io.DataIter):
+class _DALIIteratorBase(mx.io.DataIter):
+    """
+    Base class with methods shared by both DALIGenericIterator and DALIGluonIterator.
+    """
+    def __init__(self,
+                 pipelines,
+                 size,
+                 fill_last_batch,
+                 last_batch_padded,
+                 auto_reset):
+        if not isinstance(pipelines, list):
+            pipelines = [pipelines]
+        assert pipelines is not None, "Number of provided pipelines has to be at least 1"
```

Comment:
I know this was moved, but it looks like this check won't work.
If we provide None as input, `isinstance(None, list) = False`, and then we make `pipelines=[None]`. This assert is true.
We may check first if `pipelines is not None` and after wrapping in list check if its `length > 0`.

Reply:
Good catch